src/megaapi_impl.cpp

```diff
@@ -16047,6 +16058,16 @@ void MegaApiImpl::processTransferFailed(Transfer *tr, MegaTransferPrivate *trans
     transfer->setLastError(megaError);
     transfer->setPriority(tr->priority);
     transfer->setState(MegaTransfer::STATE_RETRYING);
+
+    if (e == API_EOVERQUOTA)
+    {
+        unique_ptr<MegaNode> targetNode(getNodeByHandle(tr->target));
```

Comment:
In order to avoid a deleted node for storage overquota error, it would be better to store the `mForeignTarget` at the `Trasfer`, instead of the nodehandle, which requires lookups.